src/components/infinite-scroll.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+type InfiniteScrollProps = {
+  onBottomHit: () => void;
+  isLoading: boolean;
+  children: React.ReactNode;
+};
+
+function isBottom(ref: React.RefObject<HTMLDivElement>) {
+  if (!ref.current) {
+    return false;
+  }
+  return ref.current.getBoundingClientRect().bottom <= window.innerHeight;
+}
+
+export const InfiniteScroll = (props: InfiniteScrollProps) => {
```

Comment:
```suggestion
export const InfiniteScroll: React.FC<InfiniteScrollProps> = (props) => {
```

Principalmente quando se usa o `children`, é uma boa tipar o componente dessa forma, porque isso deixa mais claro.

Reply:
Certo, mas não consegui entender muito bem a diferença